File: examples/python/send_push.py
"""Create, send and measure a Push Notification campaign (server API).

Run:  PUSHFY_PUSH_KEY=pushk_... PUSHFY_PUSH_SECRET=pss_... python3 send_push.py

The Push server API is signed with HMAC — the SDK handles the signing once you
pass push_key / push_secret.
"""

import os

from pushfy import Pushfy


def main():
    pushfy = Pushfy(
        push_key=os.environ["PUSHFY_PUSH_KEY"],
        push_secret=os.environ["PUSHFY_PUSH_SECRET"],
    )

    campaign = pushfy.push.campaigns.create(
        {
            "name": "Promo",
            "title": "Sale!",
            "body": "50% off today only",
            "url": "https://example.com/sale",
        }
    )
    campaign_id = campaign["id"]
    print("Created campaign:", campaign_id)

    pushfy.push.campaigns.send(campaign_id)
    print("Campaign sent.")

    metrics = pushfy.push.campaigns.metrics(campaign_id)
    print("Metrics:", metrics)


if __name__ == "__main__":
    main()

File: examples/typescript/send-push.ts
/**
 * send-push.ts — Create and dispatch a Push Notification campaign (server API).
 *
 * Uses the V2 Push server credentials (HMAC-signed automatically by the SDK).
 *
 * Run:  PUSHFY_PUSH_KEY=... PUSHFY_PUSH_SECRET=... npx ts-node send-push.ts
 */
import { Pushfy, JsonValue } from '@pushfy/pushfy';

/** Narrow the loosely-typed V2 JSON into the id we need. */
function readId(value: JsonValue): string {
  if (value && typeof value === 'object' && !Array.isArray(value)) {
    const id = (value as Record<string, JsonValue>).id;
    if (typeof id === 'string' || typeof id === 'number') return String(id);
  }
  throw new Error(`Unexpected campaign response: ${JSON.stringify(value)}`);
}

async function main(): Promise<void> {
  const pushKey = process.env.PUSHFY_PUSH_KEY;
  const pushSecret = process.env.PUSHFY_PUSH_SECRET;
  if (!pushKey || !pushSecret) {
    throw new Error('Set PUSHFY_PUSH_KEY and PUSHFY_PUSH_SECRET in the environment.');
  }

  const pushfy = new Pushfy({ pushKey, pushSecret });

  // 1. Create the campaign.
  const created = await pushfy.push.campaigns.create({
    name: 'Weekend Promo',
    title: 'Sale is live! 🎉',
    body: '50% off — this weekend only.',
    url: 'https://example.com/promo',
  });
  const campaignId = readId(created);
  console.log('Created campaign:', campaignId);

  // 2. Dispatch it.
  await pushfy.push.campaigns.send(campaignId);
  console.log('Campaign sent.');

  // 3. Read back delivery metrics.
  const metrics = await pushfy.push.campaigns.metrics(campaignId);
  console.log('Metrics:', JSON.stringify(metrics, null, 2));
}

main().catch((err: unknown) => {
  console.error('Failed to send push:', err instanceof Error ? err.message : err);
  process.exit(1);
});
